app/processor/string.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package processor
+
+import (
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/thedevsaddam/gojsonq"
+
+	"github.com/mendersoftware/workflows/model"
+	"github.com/mendersoftware/workflows/utils"
+)
+
+const (
+	workflowEnvVariable   = "env."
+	workflowInputVariable = "workflow.input."
+	regexVariable         = `\$\{([^\}\|]+)(?:\|([^\}]+))?}`
+	regexOutputVariable   = `(.*)\.json\.(.*)`
+)
+
+var reExpression = regexp.MustCompile(regexVariable)
+var reExpressionOutput = regexp.MustCompile(regexOutputVariable)
+
+type Encoding int64
+
+const (
+	Plain Encoding = iota
+	URL
+)
+
+type JobStringProcessor struct {
+	workflow *model.Workflow
+	job      *model.Job
+}
+
+type Options struct {
+	Encoding Encoding
+}
+
+func NewJobStringProcessor(
+	workflow *model.Workflow,
+	job *model.Job,
+) *JobStringProcessor {
+	return &JobStringProcessor{
+		workflow: workflow,
+		job:      job,
+	}
+}
+
+func (j *JobStringProcessor) ProcessJobString(data string, options *Options) string {
```

Comment:
If you use a variadic function parameter, you can save the `, nil` where you don't need otpions:

```suggestion
func (j *JobStringProcessor) ProcessJobString(data string, opts ...*Options) string {
```

At this point, you can loop over options (not tested):
```
    options := Options{Encoding: Plain}
    for _, o := range opts {
        options.Encoding = o.Encoding
    }
```

If you want to spend a bit more time, you can use the functional options pattern:
https://www.sohamkamani.com/golang/options-pattern/

Reply:
I was following your previous suggestions :) I like the `...`. the functional options: nice! I have never used it. I think `...` is nice here. is it ok now?

Reply:
will you believe it? `app/processor/string.go:67:1: cyclomatic complexity 21 of func `(*JobStringProcessor).ProcessJobString` is high (> 20) (gocyclo)` there is always something.